File: internal/dto/role.go
package dto

type RoleRef struct {
	Namespace string `json:"namespace"`
	Name      string `json:"name"`
}

type Role struct {
	Name          string
	Namespace     string
	Age           string
	CreatedAt     string
	Labels        map[string]string
	Annotations   map[string]string
	ManagedFields []ManagedField
	Rules         []PolicyRule
}

File: packages/core/frontend/src/types/resources/role.ts
import type { ManagedField } from "./shared";
import type { PolicyRule } from "./clusterrole";

export interface Role {
  Name: string;
  Namespace: string;
  Age: string;
  CreatedAt: string;
  Labels: Record<string, string>;
  Annotations: Record<string, string>;
  ManagedFields: ManagedField[];
  Rules: PolicyRule[];
}
